.github/workflows/prbot.yml

```diff
@@ -395,3 +395,19 @@ jobs:
             echo "Build is invalid"
             exit 1
           fi
+  is-valid:
+    name: 'Confirm build is valid'
+    needs:
+      - 'is-valid-pr'
+      - 'is-valid-merge-queue'
+    runs-on: ubuntu-latest
+    steps:
+      - run: |
+          success="${{ contains(needs.*.result, 'success')}}"
```

Comment:
We check that either `is-valid-pr` or `is-valid-merge-queue` is successful to return ✅ 
If neither are, 💥

Reply:
Not sure I 100% understand the change as related to the exposed problem.

But be careful, because I've had the same problem before, if a job is skipped, I _think_ it "counts" as success?
So in the case where 1 of the two `is-valid` is skipped, it would also return true I think?

Not 100% sure, but I remember having an issue myself with this same `if` in my CI :)

Reply:
Good gotcha, 'cause it's an easy trap to fell into.
https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/accessing-contextual-information-about-workflow-runs#needs-context
There's a distinction on success/failure/cancel/skipped when you evaluate them directly.
When you just depends-on and rely on GitHub, skipped is coerced as 'okay-ish' status (i.e. wont fail everything).